sql/core/src/test/scala/org/apache/spark/sql/ResolveDefaultColumnsSuite.scala

```diff
@@ -213,4 +214,29 @@ class ResolveDefaultColumnsSuite extends QueryTest with SharedSparkSession {
       }
     }
   }
+
+  test("SPARK-46949: DDL with valid default char/varchar values") {
+    withTable("t") {
+      val ddl =
+        s"""
+           |CREATE TABLE t(
+           |  key int,
+           |  v VARCHAR(6) DEFAULT 'apache',
+           |  c CHAR(5) DEFAULT 'spark')
```

Comment:
shall we do char padding for default values?

Reply:
Does padding only get involved with the read path?

Reply:
For safety reasons, I added a new test at L245.